core/src/test/java/org/mobilitydata/gtfsvalidator/type/GtfsDateTest.java

```diff
@@ -24,15 +24,78 @@
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
+import java.time.LocalDate;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertThrows;
+
 @RunWith(JUnit4.class)
 public class GtfsDateTest {
-  @Test
-  public void fromString() {
-    assertThat(GtfsDate.fromString("20200901").getLocalDate()).isEqualTo(LocalDate.of(2020, 9, 1));
-    assertThat(GtfsDate.fromString("19970103").getLocalDate()).isEqualTo(LocalDate.of(1997, 1, 3));
-
-    assertThrows(IllegalArgumentException.class, () -> GtfsDate.fromString("0"));
-    assertThrows(IllegalArgumentException.class, () -> GtfsDate.fromString("qwerty"));
-    assertThrows(IllegalArgumentException.class, () -> GtfsDate.fromString("today"));
-  }
+    @Test
+    public void fromString() {
+        assertThat(GtfsDate.fromString("20200901").getLocalDate())
+                .isEqualTo(LocalDate.of(2020, 9, 1));
+        assertThat(GtfsDate.fromString("19970103").getLocalDate())
+                .isEqualTo(LocalDate.of(1997, 1, 3));
+
+        assertThrows(IllegalArgumentException.class, () -> GtfsDate.fromString("0"));
+        assertThrows(IllegalArgumentException.class, () -> GtfsDate.fromString("qwerty"));
+        assertThrows(IllegalArgumentException.class, () -> GtfsDate.fromString("today"));
+    }
+
+    @Test
+    public void dateFromLocalDateShouldReturnGtfsDateWithSameData() {
+        GtfsDate underTest = GtfsDate.fromLocalDate(LocalDate.of(2022,1,20));
+        assertThat(underTest.getDay()).isEqualTo(20);
+        assertThat(underTest.getMonth()).isEqualTo(1);
+        assertThat(underTest.getYear()).isEqualTo(2022);
+    }
+
+    @Test
+    public void dateFromEpochShouldReturnGtfsDateWithSameData() {
+        GtfsDate underTest = GtfsDate.fromEpochDay(18614);
+        assertThat(underTest.getDay()).isEqualTo(18);
+        assertThat(underTest.getMonth()).isEqualTo(12);
+        assertThat(underTest.getYear()).isEqualTo(2020);
```

Comment:
This can be written shorter:

assertThat(GtfsDate.fromEpochDay(18614).getLocalDate()).isEqualTo(LocalDate.ofEpochDay(18614));